linalg/hypre.cpp

```diff
@@ -3017,7 +3017,8 @@ void GatherBlockOffsetData(MPI_Comm comm, const int rank, const int nprocs,
    }
 }
 
-HypreParMatrix * HypreParMatrixFromBlocks(Array2D<HypreParMatrix*> &blocks,
+HypreParMatrix * HypreParMatrixFromBlocks(Array2D<const HypreParMatrix*>
```

Comment:
Nit: This is probably better formatted as
```suggestion
HypreParMatrix * HypreParMatrixFromBlocks(
   Array2D<const HypreParMatrix*> &blocks,
```